app/queries/subscriptions_query.rb

```diff
@@ -19,4 +20,17 @@ def with_external_customer(scope)
   def with_plan_code(scope)
     scope.joins(:plan).where(plans: { code: filters.plan_code })
   end
+
+  def with_status(scope)
+    if filters.status.present? && valid_status?
+      scope.where(status: filters.status)
+    else
+      # Return active subscriptions by default
+      scope.where(status: [:active])
+    end
+  end
+
+  def valid_status?
+    filters.status.all? { |s| Subscription.statuses.key?(s) }
```

Comment:
maybe you can add `filter.status.present?` here?

Reply:
I did it outside but can definitely move it in here, that would make sense

Reply:
done